pilot_experiment/functions/calculations.py

```diff
@@ -0,0 +1,245 @@
+                                    'num_non_cooperonic_coexpressed_core': [num_non_cooperonic_coexpressed_core_genes[k] 
+                                                                            for k in core_gene_ids],
+                                    'num_non_cooperonic_coexpressed_acc': [num_non_cooperonic_coexpressed_acc_genes[k] 
+                                                                            for k in core_gene_ids],
+                                    'percent_non_cooperonic_coexpressed_core': [non_cooperonic_coexpressed_core_prop[k] 
+                                                                                for k in core_gene_ids],
+                                    'percent_non_cooperonic_coexpressed_acc': [non_cooperonic_coexpressed_acc_prop[k] 
+                                                                                for k in core_gene_ids]
+                                    }
+                                )
+
+    # Accessory gene stats
+    acc_stats_df = pd.DataFrame(data={'ref_gene':acc_gene_ids,
+                                    'num_coexpressed_genes':num_coexpressed_genes[acc_gene_ids],
+                                    'num_coexpressed_core': [num_coexpressed_core_genes[a] for a in acc_gene_ids],
+                                    'num_coexpressed_acc': [num_coexpressed_acc_genes[a] for a in acc_gene_ids],
+                                    'percent_coexpressed_core': [coexpressed_core_prop[a] for a in acc_gene_ids],
+                                    'percent_coexpressed_acc': [coexpressed_acc_prop[a] for a in acc_gene_ids],
+                                    'num_non_cooperonic_coexpressed_genes':[num_non_cooperonic_coexpressed_genes[a] 
+                                                                            for a in acc_gene_ids],
+                                    'num_non_cooperonic_coexpressed_core': [num_non_cooperonic_coexpressed_core_genes[a] 
+                                                                            for a in acc_gene_ids],
+                                    'num_non_cooperonic_coexpressed_acc': [num_non_cooperonic_coexpressed_acc_genes[a] 
+                                                                            for a in acc_gene_ids],
+                                    'percent_non_cooperonic_coexpressed_core': [non_cooperonic_coexpressed_core_prop[a] 
+                                                                                for a in acc_gene_ids],
+                                    'percent_non_cooperonic_coexpressed_acc': [non_cooperonic_coexpressed_acc_prop[a] 
+                                                                                for a in acc_gene_ids]
+                                    }
+                                )
+    return core_stats_df, acc_stats_df
+            
+
+def map_gene_to_operon(operon_annotation_file,
```

Comment:
I can change this name. Just out of curiosity, what were you expecting to be returned when you say gene: operon? I guess "operon" and "list of cooperonic genes" sounds synonymous to me..but I might be missing something

Reply:
I guess for gene:operon I would think of some kind of identifier (I don't know if operons have names, but if they do maybe this would store their names).

I don't feel too strongly about this though - I just noticed you did write in the function documentation that it maps to a list of genes, so that's probably sufficient.